addon/services/asset-loader.js

```diff
@@ -89,6 +89,12 @@ export default Ember.Service.extend({
     const assets = bundle.assets || [];
     const assetPromises = assets.map((asset) => this.loadAsset(asset, retryLoad));
 
+    // ember-auto-import creates window.__eaiEngineLookup when a lazy engine uses eai v2.
+    // this enables lazy engine's imports to be lazy themselves.
+    if (window.__eaiEngineLookup && window.__eaiEngineLookup[name]) {
+      assetPromises.push(window.__eaiEngineLookup[name]());
+    }
```

Comment:
Let's avoid specifically checking `window` (we don't really need to, and it might be wrong if ran in other contexts like service worker / node / etc).

```suggestion
    if (typeof __eaiEngineLookup === 'object' && __eaiEngineLookup[name]) {
      assetPromises.push(__eaiEngineLookup[name]());
    }
```